packages/babel-plugin-transform-typescript/src/enum.js

```diff
@@ -88,21 +90,27 @@ function enumFill(path, t, id) {
  *     Z = X | Y,
  *   }
  */
-type PreviousEnumMembers = { [name: string]: number | typeof undefined };
+type PreviousEnumMembers = { [name: string]: number | string | undefined };
 
 function translateEnumValues(path, t) {
   const seen: PreviousEnumMembers = Object.create(null);
   // Start at -1 so the first enum member is its increment, 0.
-  let prev: number | typeof undefined = -1;
+  let prev: number | undefined = -1;
   return path.node.members.map(member => {
     const name = t.isIdentifier(member.id) ? member.id.name : member.id.value;
     const initializer = member.initializer;
     let value: Expression;
     if (initializer) {
       const constValue = evaluate(initializer, seen);
       if (constValue !== undefined) {
-        value = t.numericLiteral(constValue);
-        prev = constValue;
+        seen[name] = constValue;
+        if (typeof constValue === "number") {
+          value = t.numericLiteral(constValue);
+          prev = constValue;
+        } else {
+          value = t.stringLiteral(constValue);
```

Comment:
Can you add an assertion here? Something like
```js
else if (typeof constValue === "string") {
  value = t.stringLiteral(constValue);
  prev = undefined;
} else {
  throw path.buildCodeFrameError("Invalid enum value");
}
```

To disallow things like
```ts
enum Enum {
  baz = true
}
```

Reply:
For `baz = true` we will simply not have a `constValue` -- it won't be set to anything other than a number or string. Still, added `assert(typeof constValue === "string");`.